src/crafting.cpp

```diff
@@ -538,57 +537,56 @@ void player::complete_craft()
 
     int diff_dice;
     if( secondary_difficulty > 0 ) {
-        diff_dice = making->difficulty * 3 + secondary_difficulty;
+        diff_dice = making.difficulty * 3 + secondary_difficulty;
     } else {
         // Since skill level is * 4 also
-        diff_dice = making->difficulty * 4;
+        diff_dice = making.difficulty * 4;
     }
 
     int diff_sides = 24; // 16 + 8 (default intelligence)
 
     int skill_roll = dice( skill_dice, skill_sides );
     int diff_roll  = dice( diff_dice,  diff_sides );
 
-    if( making->skill_used ) {
+    if( making.skill_used ) {
         // normalize experience gain to crafting time, giving a bonus for longer crafting
-        const double batch_mult = batch_size + base_time_to_craft( *making, batch_size ) / 30000.0;
-        practice( making->skill_used, static_cast<int>( ( making->difficulty * 15 + 10 ) * batch_mult ),
-                  static_cast<int>( making->difficulty ) * 1.25 );
+        const double batch_mult = batch_size + base_time_to_craft( making, batch_size ) / 30000.0;
+        practice( making.skill_used, static_cast<int>( ( making.difficulty * 15 + 10 ) * batch_mult ),
+                  static_cast<int>( making.difficulty ) * 1.25 );
 
         //NPCs assisting or watching should gain experience...
-        for( auto &elem : helpers ) {
+        for( auto &helper : helpers ) {
             //If the NPC can understand what you are doing, they gain more exp
-            if( elem->get_skill_level( making->skill_used ) >= making->difficulty ) {
-                elem->practice( making->skill_used,
-                                static_cast<int>( ( making->difficulty * 15 + 10 ) * batch_mult *
-                                                  .50 ), static_cast<int>( making->difficulty ) * 1.25 );
+            if( helper->get_skill_level( making.skill_used ) >= making.difficulty ) {
+                helper->practice( making.skill_used,
+                                  static_cast<int>( ( making.difficulty * 15 + 10 ) * batch_mult *
+                                                    .50 ), static_cast<int>( making.difficulty ) * 1.25 );
                 if( batch_size > 1 ) {
-                    add_msg( m_info, _( "%s assists with crafting..." ), elem->name.c_str() );
+                    add_msg( m_info, _( "%s assists with crafting..." ), helper->name );
```

Comment:
You can cache `helper->name` in variable and reference it whenever needed.

Reply:
We could, but I don't think there's any particular reason to.  It would only be replacing `helper->name` with `helper_name`.